core/src/components/datetime/datetime.tsx

```diff
@@ -2401,7 +2437,7 @@ export class Datetime implements ComponentInterface {
             [`datetime-presentation-${presentation}`]: true,
             [`datetime-size-${size}`]: true,
             [`datetime-prefer-wheel`]: hasWheelVariant,
-            [`datetime-grid`]: hasGrid,
+            [`datetime-grid`]: this.isGridStyle,
```

Comment:
~ nit: Can we refactor this to `isGridStyle` to match the code style (destructure from `this`)?